hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/SparkMergeHelper.java

```diff
@@ -77,14 +89,39 @@ public void runMerge(HoodieTable<T, JavaRDD<HoodieRecord<T>>, JavaRDD<HoodieKey>
       readSchema = mergeHandle.getWriterSchemaWithMetaFields();
     }
 
+    Option<InternalSchema> querySchemaOpt = SerDeHelper.fromJson(table.getConfig().getInternalSchema());
+    Boolean needToReWriteRecord = false;
+    // to do support bootstrap
+    if (querySchemaOpt.isPresent() && !baseFile.getBootstrapBaseFile().isPresent()) {
+      // check implicitly add columns, and position reorder(spark sql may change cols order)
+      InternalSchema querySchema = AvroSchemaUtil.evolutionSchemaFromNewAvroSchema(readSchema, querySchemaOpt.get(), true);
+      long commitTime = Long.valueOf(FSUtils.getCommitTime(mergeHandle.getOldFilePath().getName()));
+      InternalSchema writeInternalSchema = TableInternalSchemaUtils.searchSchemaAndCache(commitTime, table.getMetaClient());
+      List<String> colNamesFromQuerySchema = querySchema.getAllColsFullName();
+      List<String> colNamesFromWriteSchema = writeInternalSchema.getAllColsFullName();
+      List<String> diffCols = colNamesFromWriteSchema.stream()
```

Comment:
Should this be named sameCols?

Reply:
yes，you are right,  it should be named sameCols

Reply:
fixed